### PR TITLE
spark: build packages for supported JDK versions

### PR DESCRIPTION
**Pull request by Dentrax**
Allows us to bundle the correct JDK version with the package and run tests for each supported JDK version at the package level for Spark.

### CODE REVIEW

**Review by EyeCantCU — CHANGES_REQUESTED** [review]

File: spark-3.5.yaml [hunk 1]
```diff
@@ -63,18 +65,101 @@ pipeline:
...
-      mv dist/* ${{targets.destdir}}/usr/lib/spark/
+          ./dev/make-distribution.sh --name custom-spark --pip -Psparkr -Phive -Phive-thriftserver -Pmesos -Pyarn -Pkubernetes
+
+          patch dist/bin/load-spark-env.sh load-spark-env.sh.diff
+          patch dist/sbin/spark-daemon.sh spark-daemon.sh.diff
+
+          mkdir -p ${{targets.contextdir}}/usr/lib/spark
+          mkdir -p ${{targets.contextdir}}/usr/lib/spark/work-dir
+          mv dist/* ${{targets.contextdir}}/usr/lib/spark/
+    test:
```
[the remark this answers is not in the document]
Comment (Dentrax) [reply spark-3.5.yaml:94]: Hmmm actually I'm including the testing deps in the `environment.contents.packages` using `${{range.value}}`. 🤔

[the remark this answers is not in the document]
Comment (Dentrax) [reply spark-3.5.yaml:94]: I'm going to resolve this conversation as tests are now passing, feel free to re-open if you have further thoughts on this!

File: spark-3.5.yaml [hunk 2]
```diff
@@ -63,18 +62,105 @@ pipeline:
...
 
-  - name: build
-    runs: |
-      ./dev/make-distribution.sh --name custom-spark --pip -Psparkr -Phive -Phive-thriftserver -Pmesos -Pyarn -Pkubernetes
+data:
+  - name: openjdk-versions
+    items:
+      8: "java-1.8-openjdk"
+      11: "java-11-openjdk"
+      17: "java-17-openjdk"
```
Comment (EyeCantCU) [inline spark-3.5.yaml:70]: Does Spark run with OpenJDK 21? If so, we should add it here

Comment (Dentrax) [reply spark-3.5.yaml:70]: I don't think they "official" have support on `21`:

https://spark.apache.org/docs/latest/
> Spark runs on Java 8/11/17

And also no Java 21 on their [versions.json](https://github.com/apache/spark-docker/blob/master/versions.json).

**Review by EyeCantCU — APPROVED** [review]

**Pull request merged**